x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTruncProcessorTests.java

```diff
@@ -129,6 +143,91 @@ public void testTruncation() {
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+2019-10",
+            toString((IntervalYearMonth) new DateTrunc(Source.EMPTY, l("weeks"), yearToMonth, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+2019-10",
+            toString((IntervalYearMonth) new DateTrunc(Source.EMPTY, l("days"), yearToMonth, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+2019-10",
+            toString((IntervalYearMonth) new DateTrunc(Source.EMPTY, l("hh"), yearToMonth, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+2019-10",
+            toString((IntervalYearMonth) new DateTrunc(Source.EMPTY, l("mi"), yearToMonth, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+2019-10",
+            toString((IntervalYearMonth) new DateTrunc(Source.EMPTY, l("second"), yearToMonth, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+2019-10",
+            toString((IntervalYearMonth) new DateTrunc(Source.EMPTY, l("ms"), yearToMonth, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+2019-10",
+            toString((IntervalYearMonth) new DateTrunc(Source.EMPTY, l("mcs"), yearToMonth, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+2019-10",
+            toString((IntervalYearMonth) new DateTrunc(Source.EMPTY, l("nanoseconds"), yearToMonth, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+0 00:00:00.0",
+            toString((IntervalDayTime) new DateTrunc(Source.EMPTY, l("millennia"), dayToSecond, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+0 00:00:00.0",
+            toString((IntervalDayTime) new DateTrunc(Source.EMPTY, l("CENTURY"), dayToSecond, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+0 00:00:00.0",
+            toString((IntervalDayTime) new DateTrunc(Source.EMPTY, l("decades"), dayToSecond, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+0 00:00:00.0",
+            toString((IntervalDayTime) new DateTrunc(Source.EMPTY, l("years"), dayToSecond, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+0 00:00:00.0",
+            toString((IntervalDayTime) new DateTrunc(Source.EMPTY, l("quarters"), dayToSecond, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+0 00:00:00.0",
+            toString((IntervalDayTime) new DateTrunc(Source.EMPTY, l("month"), dayToSecond, null)
+                .makePipe().asProcessor().process(null)));
+        assertEquals("+0 00:00:00.0",
+            toString((IntervalDayTime) new DateTrunc(Source.EMPTY, l("weeks"), dayToSecond, null)
```

Comment:
Same here:
```
postgres=# select date_trunc('weeks', INTERVAL '105 DAYS 2 HOURS 45 MINUTES 55 SECONDS');
ERROR:  interval units "weeks" not supported because months usually have fractional weeks```